symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/command/CommandActivity.java

```diff
@@ -53,10 +59,23 @@ public void onMessageSent(V4Initiator initiator, V4MessageSent event) {
   /**
    * {@inheritDoc}
    */
+  @SneakyThrows
   protected void beforeMatcher(C context) {
     try {
-      context.setTextContent(
-          PresentationMLParser.getTextContent(context.getSourceEvent().getMessage().getMessage()));
+      final String textContent =
+          PresentationMLParser.getTextContent(context.getSourceEvent().getMessage().getMessage());
+      context.setTextContent(textContent);
+
+      // Fails if message contains a word beginning with @ without being an actual mention
+      List<String> mentionsText = Arrays.stream(textContent.split("\\s+")).filter(s -> s.startsWith("@")).collect(Collectors.toList());
```

Comment:
what if the mention has a space in between @Yinan LIU,  it is a mention, if i understand well, this will only put @Yinan in the list of string, no?

Reply:
we have the same feature already implemented in Harmony framework, waiting for this done in BDK, so that we can remove the same from HFK. a bit shame what we cannot communicate the features shared/done in one or the other.

Reply:
> what if the mention has a space in between @Yinan LIU, it is a mention, if i understand well, this will only put @Yinan in the list of string, no?

Good point @yinan-symphony 👍 

@symphony-elias you might need to use [MessageParser#getMentions](https://github.com/finos/symphony-bdk-java/blob/main/symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/util/MessageParser.java#L36)

Reply:
Indeed, and pitfalls of this implementation have already been described here: https://perzoinc.atlassian.net/wiki/spaces/DevX/pages/2306802587/PLAT-11535+Slash+commands+arguments
The difficulty here is:
- AntPathMatcher follows the pattern we pass to it. It means we have to rewrite it so that it matches our needs
- A word beginning with @ may not be always an actual mention
- FYI when we check for bot mentions, we currently check for the text content only and not the data.

Reply:
@symphony-elias in HFK, i had to read the entity data from the presentationml and match the mention there to the message text to extract the mention name, so far is working well.

Reply:
https://github.com/SymphonyOSF/harmony-framework/blob/master/harmony-activities/src/main/java/com/symphony/harmonyframework/activities/command/MentionsReader.java